lazy-imports-import.html

```diff
@@ -22,23 +23,26 @@
       var imports = fromElement.querySelectorAll(query);
       var promises = [];
       for (var i = 0; i < imports.length; i++) {
-        promises.push(new Promise(function(resolve) {
-          var href = imports[i].getAttribute('href');
-          var onLoad = function() {
-            resolve({loaded: href});
-          };
-          var onFailure = function() {
-            resolve({failed: href});
-          };
-          var resolved = callingElement.resolveUrl(href);
-          if (!__loadedImports[resolved]) {
-            __loadedImports[resolved] = true;
+        var href = imports[i].getAttribute('href');
+        var resolvedUrl = callingElement.resolveUrl(href);
+
+        if (__failedImports[resolvedUrl] || !__loadedImports[resolvedUrl]) {
```

Comment:
I think you can get the same retry behavior while storing promises by adding a .catch() to the existing promise, then storing the result as the pending promise.

Something like this rough sketch:

```js
let promise;
if (__importPromises.hasOwnProperty(resolvedUrl)) {
  promise = __importPromises[resolvedUrl];
  if (opts && opts.retry) {
    promise = promise.catch(function(e) {
      return Polymer.__importLazyGroup(callingElement, group, fromElement, {retry: false});
    });
  }
} else {
  promise = new Promise(...);
}
__importPromises[resolvedUrl] = promise;
return promise;
```

This would make retry work in the presence of a pending request that will fail. I guess it depends on the desired semantics for retry.